server/utils.py

```diff
@@ -64,28 +63,41 @@ def server_time_obj(time, course):
     return time.astimezone(pytz.utc)
 
 def future_time_obj(course, **kwargs):
-    """Get a datetime object representing some timedelta from now with the clock
-    time set at 23:59:59."""
+    """ Get a datetime object representing some timedelta from now with the time
+    set at 23:59:59.
+    """
     date = course.timezone.localize(dt.datetime.now() + dt.timedelta(**kwargs))
     time = dt.time(hour=23, minute=59, second=59, microsecond=0)
     return dt.datetime.combine(date, time)
 
 def new_due_date(course):
-    """Return a string representing a new due date next week."""
+    """ Return a string representing a new due date next week."""
     return future_time_obj(course, weeks=1).strftime(constants.ISO_DATETIME_FMT)
 
 def new_lock_date(course):
-    """Return a string representing a new lock date 8 days from now."""
+    """ Return a string representing a new lock date 8 days from now."""
     return (future_time_obj(course, weeks=1, days=1)
             .strftime(constants.ISO_DATETIME_FMT))
 
 def natural_time(date):
-    """Format a human-readable time difference (e.g. "6 days ago")"""
+    """ Format a human-readable time difference (e.g. "6 days ago")"""
     if date.tzinfo:
         date = date.astimezone(pytz.utc).replace(tzinfo=None)
     now = dt.datetime.utcnow()
     return humanize.naturaltime(now - date)
 
+
+def humanize_name(name):
+    """ Return a canonical representation of a name in First Last format."""
+    parts = [part.strip() for part in name.split(",")]
+    if name == name.upper():
+        split = [["-".join(subword.capitalize() for subword in word.split("-"))
+                    for word in part.split()]
+                for part in parts]
+        parts = [" ".join(part) for part in split]
+    return " ".join(parts[::-1])
```

Comment:
Yeah, I found that too. I tested `humanize_name` against the entire fa16 roster and they looked good to my eye. But the bCourses dataset is a little limited in that regard since there's only up two parts (1 comma separating first name from the last name) and no rare suffixes like Jr. in this semester's class.